trio/_subprocess.py

```diff
@@ -171,7 +172,11 @@ def __init__(
             else:
                 # It worked! Wrap the raw fd up in a Python file object to
                 # make sure it'll get closed.
-                self._pidfd = open(fd)
+                # fmt: off
+                self._pidfd = open(  # noqa: SIM115  # open-file-with-context-handler
```

Comment:
I think ruff shouldn't error on this case, this is just silly. Not every class should be constructable solely using a context manager.

Reply:
Yeah this one could be reported upstream. Why is the `# fmt: off` needed ?

Reply:
`# fmt: off` needed because basically ruff and black end up fighting over the formatting if I remember correctly. I think it's because of differences in the way the two handle trying to make it not exceed a line number limit.

Reply:
that *also* sounds like something that should be reported upstream :sweat_smile: 

if we're `fmt: off`ing then I think `fd)` should be moved onto line 176, kinda silly to split it up this much.